### DISCUSSION TITLE
Applicability of PR #10129 for Kong 2.5.1: Concerns over sequence adjustment in Kong.init_worker function

### DISCUSSION THREAD

**Discussion by dingjiayi** [Kong/kong#10977] [Help] [answered]
I'm currently trying to fix an issue on Kong 2.5.1 and have referred to  PR https://github.com/Kong/kong/pull/10129. However, I am unsure whether this PR is applicable to Kong 2.5.1 due to the following concerns: 
The reason for adjusting the position of the call to runloop.init_worker.before() in the Kong.init_worker function and whether this sequence adjustment can also be applied to Kong 2.5.1.

Thank you.

**Comment by mureo** [comment]
We have also encountered the same problem, but we are unable to upgrade Kong to the fixed version in the short term. Please help

**Comment by chensunny** [comment]
@cheald Can it be merged into 2.5.x release or 2.8.x release, we use 2.5.2 but cannot upgrade 2.8.x.

**Comment by dingjiayi** [comment]
@ms2008 May I kindly request your suggestions and help ? Thank you.

**Comment by ms2008** [comment, marked as answer]
> The reason for adjusting the position of the call to `runloop.init_worker.before()` in the `Kong.init_worker` function

`runloop.init_worker.before()` will create two timers to rebuild router and rebuild plugins iterator respectively, the rebuild process will first compare the corresponding `*:version` in the cache whether it is the latest, If it is up to date, then there is no need to rebuild it to improve performance.

But when init_worker is just started we manually execute `runloop.update_plugins_iterator()` and `runloop.update_router()` to make sure they are loaded as soon as possible, obviously `runloop.init_worker. before()` asynchronous calls would be more appropriate after them.

> whether this sequence adjustment can also be applied to Kong 2.5.1.

I think there is no problem, although I have not personally verified.

**Reply by dingjiayi** [reply]
Thank you very much.
